backend/src/flotilla/api/robots_api.py

```diff
@@ -46,14 +52,124 @@ async def read_robots(response: Response, db: Session = Depends(get_db)) -> List
     summary="Lookup a single robot",
     dependencies=[Security(authentication_scheme)],
 )
-async def read_robot(
+async def get_robot(
     response: Response,
     robot_id: int = Path(None, description=""),
     db: Session = Depends(get_db),
 ) -> Robot:
-    robot: RobotDB = get_robot_by_id(db=db, robot_id=robot_id)
+    robot: RobotDB = read_robot_by_id(db=db, robot_id=robot_id)
     if not robot:
         response.status_code = 404
         return ProblemDetails(title=f"No robot with id: {robot_id}", status=404)
     robot_api: Robot = robot.get_api_robot()
     return robot_api
+
+
+@router.post(
+    "/robots/{robot_id}/start/{mission_id}",
+    responses={
+        200: {"model": StartResponse, "description": "Mission succesfully started"},
+        403: {
+            "model": ProblemDetails,
+            "description": "Failed to start - Client is not authorized to start mission",
+        },
+        409: {
+            "model": ProblemDetails,
+            "description": "Failed to start - Conflict with the robot schedule",
+        },
+        404: {
+            "model": ProblemDetails,
+            "description": "Not Found - No robot with given id",
+        },
+        502: {
+            "model": ProblemDetails,
+            "description": "Failed to start - Error while contacting ISAR",
+        },
+    },
+    tags=["Robots"],
+    summary="Start a mission with robot",
+    dependencies=[Security(authentication_scheme)],
+)
+async def post_start_robot(
+    response: Response,
+    robot_id: int = Path(None, description=""),
+    mission_id: int = Path(None, description=""),
+    db: Session = Depends(get_db),
+    isar_requests: IsarRequests = Depends(get_isar_requests),
+) -> StartResponse:
+    """### Overview Start a mission with given id using robot with robot id."""
+    robot: RobotDB = read_robot_by_id(db, robot_id)
+    if not robot:
+        response.status_code = 404
+        return ProblemDetails(title=f"Not found - No robot with id {robot_id} exists")
+
+    try:
+        response_isar: RequestResponse = isar_requests.start_mission(
+            host=robot.host, port=robot.port, mission_id=mission_id
+        )
+    except RequestException:
+        response.status_code = 502
+        return ProblemDetails(title="Error sending request to ISAR")
+    response_isar_json: dict = response_isar.json()
+    if response_isar.status_code != 200:
+        response.status_code = response_isar.status_code
+        return ProblemDetails(
+            title=f"Could not start mission. Reason: {response_isar_json['message']}"
+        )
```

Comment:
This should not be necessary. If you look at the `RequestHandler` it uses a function `response.raise_for_status()`. This will raise an error for any status code outside the 200 range and as such we should not get to this if statement as it will have been raised as a `RequestException` in `RequestHandler`

Reply:
Should we change this behavior? Seems like we can loose a lot of useful information from using the raise_from_status, like if there is a conflict. Don't know if we would handle it any different, but might want to pass the information about the particular issue?

Reply:
The error message itself is kept as we use `raise error from e`. As such we should only need to log the error to have it available internally. Typically you would like to not send all the error information to the frontend but rather log it in the backend itself. As such I think this should be fine as we are saving the information in the error itself.
https://github.com/equinor/flotilla/pull/63/files#diff-c5827550d464dc4af6f269546e3cd1aae61565d50740f7837449c79f12ab8436R54-R58

Reply:
Think the try catch around the raise_for_status call should be removed and let the error be handled in the controller.

Reply:
But this should probably be done in a separate issue.

Reply:
Sure, that may be more correct for this case.